consensus/istanbul/backend/announce.go

```diff
@@ -260,8 +264,15 @@ func (sb *Backend) handleIstAnnounce(payload []byte) error {
 
 	logger = logger.New("msgAddress", msg.Address, "msg_timestamp", announceData.Timestamp)
 
+	// Check to see if the msg's timestamp is within 20 seconds of this computer's timestamp (This is the same clock threshold check that the p2p layer uses)
+	now := time.Now()
+	if int64(announceData.Timestamp.Uint64()) < now.Add(-timestampDiffThreshold).Unix() || int64(announceData.Timestamp.Uint64()) > now.Add(timestampDiffThreshold).Unix() {
+		logger.Warn("Ignoring announce messages since it's timestamp diff is over the threshold", "now", now, "announce timestamp", announceData.Timestamp.Uint64())
```

Comment:
nits: maybe `Info`? also `it's` -> `its` and maybe more useful to log the diff in seconds so you can e.g graph them or add them to metrics more easily